genomesubsampler/prototypeSelection.py

```diff
@@ -180,28 +238,32 @@ def prototype_selection_constructive_maxdist(dm, num_prototypes):
     def prototype_selection_constructive_maxdist(dm: DistanceMatrix,
     num_prototypes: int) -> List[str]:
     '''
-    if num_prototypes < 2:
-        raise ValueError(("'num_prototypes' must be >= 2, since a single "
-                          "prototype is useless."))
-    if num_prototypes >= len(dm.ids):
-        raise ValueError(("'num_prototypes' must be smaller than the number of"
-                          " elements in the distance matrix, otherwise no "
-                          "reduction is necessary."))
+    _validate_parameters(dm, num_prototypes, seedset)
 
     # initially mark all elements as uncovered, i.e. as not being a prototype
     uncovered = np.asarray([np.True_] * dm.shape[0])
+    res_set, num_found_prototypes = [], 0
+
+    if len(seedset):
+        # mark elements in the seedset as found
+        seedset = set(seedset)
```

Comment:
I just deleted my last comment, I misread the code. Casting to `set` makes sense because of the lookup performed on line 244. If `seedset` is a `list` or `tuple` then it will break down to a linear search. On the assumption that `seedset` may realistically contain more than like 10 entries, it is probably better to just cast to `set` to take advantage of `O(1)` membership testing